### PR TITLE
ci: Add Solidity compatibility check

### PR DESCRIPTION
**Pull request by samuelburnham**
## Solidity-verifier compatibility
- Checks Solidity compatibility on every PR commit using the templating engine from #341 
- Fails immediately if the Rust test or overall workflow causes an error
- Comments on the PR if the Solidity test fails
- Opens an issue in https://github.com/lurk-lab/solidity-verifier if a failing commit is merged
 
This surfaces issues early and lets authors/reviewers decide how to address breaking changes, solving the CI portion of #337.

> [!NOTE]
> An earlier version of this PR did not recommend making this workflow a required status check due to an incorrect design. The current version only errors when the Rust or basic workflow steps fail, which they never should. This would be properly caught by making the workflow a required status check.

Successful runs shown in below [comment](https://github.com/lurk-lab/arecibo/pull/363#issuecomment-1993222523) and [issue](https://github.com/lurk-lab/solidity-verifier/issues/66)

### CODE REVIEW

**Review by huitseeker — DISMISSED** [review]
This looks great!

**Review by storojs72 — CHANGES_REQUESTED** [review]
Discussed issues with @samuelburnham in DM.

The CI compatibility job needs to be modified - so that it runs compatibility tests in Rust (only for IPA currently), dynamically generates unit tests for Solidity, copies those tests into fetched solidity-verifier directory and finally runs them. Rust <-> Solidity Incompatibility over particular building block can be detected by failure of dynamically generated (by Rust) and executed (in Solidity) unit-test

**Review by storojs72 — APPROVED** [review]
LGTM

File: .github/workflows/solidity.yml [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+        with:
+          repository: lurk-lab/ci-workflows
+      - uses: ./.github/actions/ci-env
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@stable
+      - uses: taiki-e/install-action@nextest
+      - uses: Swatinem/rust-cache@v2
+      - name: Run Solidity test generator
+        run: |
+          cargo nextest run -E 'test(test_solidity_compatibility)' --release --run-ignored all --nocapture > test-output
```
[the remark this answers is not in the document]
Comment (samuelburnham) [reply .github/workflows/solidity.yml:46]: Yes, documented

**Review by storojs72 — APPROVED** [review]
Excellent! Thanks

**Review by huitseeker — APPROVED** [review]

**Pull request merged**